python-backend/app/api/mines/variances/resources/variance.py

```diff
@@ -0,0 +1,135 @@
+                        trim=True,
+                        help='A note to include on the variance. Limited to 300 characters.')
+    parser.add_argument('issue_date',
+                        store_missing=False,
+                        trim=True,
+                        help='The date on which the variance was issued.')
+    parser.add_argument('received_date',
+                        store_missing=False,
+                        trim=True,
+                        help='The date on which the variance was received.')
+    parser.add_argument('expiry_date',
+                        store_missing=False,
+                        trim=True,
+                        help='The date on which the variance expires.')
+
+
+    @api.doc(
+        description=
+        'Get a list of all variances for a given mine.',
+        params={
+            'mine_guid': 'guid of the mine for which to fetch variances'
+        }
+    )
+    @requires_role_mine_view
+    @api.marshal_with(variance_model, code=200, envelope='records')
+    def get(self, mine_guid):
+        try:
+            variances = Variance.find_by_mine_guid(mine_guid)
+        except DBAPIError:
+            raise BadRequest('Invalid mine_guid')
+
+        if variances is None:
+            raise BadRequest('Unable to fetch variances')
+
+        return variances
+
+
+    @api.doc(
+        description=
+        'Create a new variance for a given mine.',
+        params={
+            'mine_guid': 'guid of the mine with which to associate the variances'
+        }
+    )
+    @api.expect(parser)
+    @requires_role_mine_create
+    @api.marshal_with(variance_model, code=200)
+    def post(self, mine_guid):
+        data = VarianceResource.parser.parse_args()
+        compliance_article_id = data['compliance_article_id']
+
+        if not compliance_article_id:
+            raise BadRequest('Must provide compliance_article_id')
+
+        variance = Variance.create(
+            compliance_article_id,
+            mine_guid,
+            # Optional fields
+            note=data.get('note'),
+            issue_date=data.get('issue_date'),
+            received_date=data.get('received_date'),
+            expiry_date=data.get('expiry_date'))
+
+        if not variance:
+            raise BadRequest('Error: Failed to create variance')
+
+        variance.save()
+        return variance
+
+
+class VarianceResource(Resource, UserMixin, ErrorMixin):
+    @api.doc(
+        description=
+        'Get a single variance.',
+        params={
+            'mine_guid': 'guid of the mine to which the variance is associated',
+            'variance_id': 'ID of the variance to fetch'
+        }
+    )
+    @requires_role_mine_view
+    @api.marshal_with(variance_model, code=200)
+    def get(self, mine_guid, variance_id):
+        try:
+            variance = Variance.find_by_variance_id(variance_id)
+        except DBAPIError:
+            raise BadRequest('Invalid variance_id')
```

Comment:
we used to try catch the query helpers... and now that i'm seeing this come up..... do you think this is better? 

```    @classmethod
    def find_by_mine_guid(cls, _id):
        try:
            uuid.UUID(_id, version=4)
            return cls.query.filter_by(mine_guid=_id).filter_by(deleted_ind=False).first()
        except ValueError:
            return None``` 

This also stops the db from having to throw an exception and the python catches it... i'm open for either. just wanted you thought.

Reply:
Yes, I think that approach is better. I think safely returning a `None` and then doing the `None` check in the resource makes plenty of sense

Reply:
Yea, ok, let's try and re-establish that pattern of using the model search helpers to check for guid formatting.